_test/test_utilities_herbert/test_str_make_cellstr.m

```diff
@@ -0,0 +1,146 @@
+        c2
+        c3
+        s1
+        s2
+        s3
+        charr
+        carr
+        sarr
+        
+        carr_deblank
+        carr_trim
+    end
+    
+    methods
+        %-------------------------------------------------------------------
+        function S = test_str_make_cellstr (name)
+            S@TestCase(name);
+            
+            % Make some 1D character arrays
+            S.ch1 = 'Hello     ';
+            S.ch2 = 'Mister Man';
+            S.ch3 = '  Yellow  ';
+            
+            % Make some cell arrays with a single 1D character array each
+            S.c1 = cellstr(S.ch1);
+            S.c2 = cellstr(S.ch2);
+            S.c3 = cellstr(S.ch3);
+            
+            % Matlab strings
+            S.s1 = string(S.ch1);
+            S.s2 = string(S.ch2);
+            S.s3 = string(S.ch3);
+            
+            % Arrays
+            S.charr = [S.ch1; S.ch2; S.ch3];    % 2D character array
+            S.carr = {S.ch1, S.ch2, S.ch3};     % 1x3 cell array
+            S.sarr = [S.s1, S.s2, S.s3];        % 1x3 string array
+            
+            S.carr_deblank = deblank(S.carr');  % column
+            S.carr_trim = strtrim(S.carr');     % column
+        end
+        
+        %-------------------------------------------------------------------
+        function test_charArray (S)
+            % Tests that a character array is stripped of trailing blanks
+            [ok,cout] = str_make_cellstr(S.ch1);
+            assertTrue(ok)
+            assertEqual(cout, {deblank(S.ch1)})
+        end
+        
+        %-------------------------------------------------------------------
+        function test_multipleCharArrays (S)
+            % Tests that output from multiple character array input is a column
+            % cell array, and stripped of trailing (but not leading) blanks
+            [ok,cout] = str_make_cellstr(S.ch1, S.ch2, S.ch3);
+            assertTrue(ok)
+            assertEqual(cout, S.carr_deblank)
+        end
+        
+        %-------------------------------------------------------------------
+        function test_multipleCharArrays_stringArray_cellstr (S)
+            % Test that multiple forms of input, and that string arrays and
+            % cellstr are NOT stripped of trailing blanks, as Matlab behaviour
+            % dictates
+            [ok,cout] = str_make_cellstr([S.ch2; S.ch3], S.sarr, S.carr);
+            assertTrue(ok)
+            assertEqual(cout, [S.carr_deblank(2:3); S.carr'; S.carr'])
+        end
+        
+        %-------------------------------------------------------------------
+        function test_emptyTypes (~)
+            % Test that empty character arrays, cellstr and strings result in
+            % empty character arrays in a cell array (essentially a test of type
+            % conversion)
+            [ok,cout] = str_make_cellstr('', {}, {''}, "");
+            assertTrue(ok)
+            assertEqual(cout, {''; ''; ''})
```

Comment:
does it really lose an input? counterintuitive behaviour

Reply:
{} is an empty cell array, so contains nothing at all. That's different to  {''} which contains an empty string. Matlab's cellstr function returns cellstr({}) as 0x0 cell array, so when concatenated with another cell array, it just disappears. My mind has been moulded by Matlab, so it's OK for me, but it may be worth remembering counterintuitive is Matlab's middle name. Remember how a scalar is also a row vector and a column vector ?